### PR TITLE
hw 4 is done

### PR DESCRIPTION
**Pull request by ha020285mdv**
review please

### CODE REVIEW

File: myapp/models.py [hunk 1]
```diff
@@ -137,8 +141,14 @@ class Comment(models.Model):
     def __str__(self):
         return f"{self.author}'s comment for '{self.liked_object}': {self.comment[:15]}"
 
+    def save(self, **kwargs):
+        if not self.id:
+            self.date = timezone.now() - timedelta(days=365)
```
Comment (PonomaryovVladyslav) [inline myapp/models.py]: Ok, what if the year is a leap?) (2016, 2020, 2024, etc?)

File: myapp/models.py [hunk 2]
```diff
@@ -137,8 +141,14 @@ class Comment(models.Model):
...
+        super().save(**kwargs)
+
     class Meta:
         verbose_name = '3_Comment'
+        ordering = ["-id"]
```
Comment (PonomaryovVladyslav) [inline myapp/models.py:162]: What is the reason for this ordering? Or it's just for a try?)

File: myapp/views.py [hunk 3]
```diff
@@ -31,3 +35,52 @@ def uuid(request, uuid):
...
+    # 1. Получить 5 последних написанных комментариев (именно текст)
+    content['task1'] = Comment.objects.order_by('-date')[:5]
+
+    # 2. Создать 5 комментариев с разным текстом, Хотя бы один должен начинаться со слова "Start",
+    #    хоть один в середине должен иметь слово "Middle", хоть один должен заканчиваться словом "Finish".
+
+    # comment = Comment(comment='Comment to test', author_id='4', content_type_id='15', object_id='2')
+    # comment.save()
+
+    # objs = Comment.objects.bulk_create([
```
Comment (PonomaryovVladyslav) [inline myapp/views.py:54]: Good solution for this task

File: myapp/views.py [hunk 4]
```diff
@@ -31,3 +35,52 @@ def uuid(request, uuid):
...
+    # Comment.objects.filter(comment__iendswith='finish').update(comment='FINISH')
+    content['task4'] = 'done'
+
+    # 5. Удалить все комментарии у которых в тексте есть буква "k", но не удалять если есть буква "с".
+    # Comment.objects.filter(comment__icontains='k').exclude(comment__icontains='c').delete()
+    content['task5'] = 'done'
+
+    # 6. Получить первые 2 комментария по дате создания к статье у которой имя автора последнее по алфавиту.
+    author = Profile.objects.order_by('name').last()
+    article = Article.objects.filter(author=author).last() #взял самую "свежую" статью автора
```
Comment (PonomaryovVladyslav) [inline myapp/views.py]: The first 2 comments can be in the different articles) Maybe unclear task, but I need 2 first comments where articles were written by the author with the last name by ABC.

**Review by PonomaryovVladyslav — CHANGES_REQUESTED** [review]
Leap year, and first 2 comments should be a bit fixed

File: myapp/views.py (hunk 4 above)
Comment (ha020285mdv) [reply myapp/views.py]: as I understood, the deal is only in the correct ordering script?
                    Comment.objects.order_by('date', '-article__author')[:2]
but if we have a timestamp in the 'date' field, the next level of sorting doesn't help a lot  
maybe I misunderstand the task again...

File: myapp/models.py (hunk 2 above)
Comment (ha020285mdv) [reply myapp/models.py:162]: only for more convenient tracking changes in the admin panel (no needed scroll after F5)

File: myapp/models.py (hunk 1 above)
Comment (ha020285mdv) [reply myapp/models.py]: I changed the way years subtraction
but, if there are no strong requirements, maybe '-timedelta(days=365.25)' can fit as well (in order to be more laconic in code)?

File: myapp/views.py [hunk 5]
```diff
@@ -77,10 +78,7 @@ def homework4(request):
     content['task5'] = 'done'
 
     # 6. Получить первые 2 комментария по дате создания к статье у которой имя автора последнее по алфавиту.
-    author = Profile.objects.order_by('name').last()
-    article = Article.objects.filter(author=author).last() #взял самую "свежую" статью автора
-    task6 = article.comments.all().order_by('date')[:2]
-
-    content['task6'] = task6
+    content['task6'] = Comment.objects.filter(content_type_id=15).order_by('date', '-article__author')[:2]
```
Comment (PonomaryovVladyslav) [inline myapp/views.py]: Hardcoding values is a bad way in 99% of situations. 15 is a hardcoding

File: myapp/models.py [hunk 6]
```diff
@@ -142,8 +143,18 @@ def __str__(self):
         return f"{self.author}'s comment for '{self.liked_object}': {self.comment[:15]}"
 
     def save(self, **kwargs):
+        def delta_years(dt, years):
+            try:
```
Comment (PonomaryovVladyslav) [inline myapp/models.py:147]: Okay, possible)

**Review by PonomaryovVladyslav — CHANGES_REQUESTED** [review]

File: myapp/views.py (hunk 5 above)
Comment (ha020285mdv) [reply myapp/views.py]: I haven't noticed, sorry...
more than 1 hour of googling and the issue is fixed)

**Review by PonomaryovVladyslav — APPROVED** [review]

**Pull request merged**